superset/assets/javascripts/dashboard/Dashboard.jsx

```diff
@@ -195,6 +199,13 @@ export function dashboardContainer(dashboard, datasources, userid) {
       return f;
     },
     addFilter(sliceId, col, vals, merge = true, refresh = true) {
+      // If slice doesn't exist, remove the related parameters in preselect_filters.
```

Comment:
this function calls `this.updateFilterParamsInUrl` 3 times. Perhaps clearer logic would be an if statement that validates that there is a slice and column, put all the logic within that block, and call `this.updateFilterParamsInUrl` once at the very end